lib/datadog/core/remote/client.rb

```diff
@@ -108,16 +108,32 @@ def sync
 
         private
 
-        def payload
+        def payload # rubocop:disable Metrics/MethodLength
           state = repository.state
 
+          client_tracer_tags = [
+            "platform:#{native_platform}", # native platform
+            # "asm.config.rules:#{}", # TODO: defined|undefined
+            # "asm.config.enabled:#{}", # TODO: true|false|undefined
+            "ruby.tracer.version:#{Core::Environment::Identity.tracer_version}",
+            "ruby.runtime.platform:#{RUBY_PLATFORM}",
+            "ruby.runtime.version:#{RUBY_VERSION}",
+            "ruby.runtime.engine.name:#{RUBY_ENGINE}",
+            "ruby.runtime.engine.version:#{ruby_engine_version}",
+            "ruby.rubygems.platform.local:#{Gem::Platform.local}",
+            "ruby.gem.libddwaf.version:#{gem_spec('libddwaf').version}",
+            "ruby.gem.libddwaf.platform:#{gem_spec('libddwaf').platform}",
+            "ruby.gem.libdatadog.version:#{gem_spec('libdatadog').version}",
+            "ruby.gem.libdatadog.platform:#{gem_spec('libdatadog').platform}",
+          ]
+
           client_tracer = {
```

Comment:
We could probably hoist `client_tracer` altogether since between requests the values would be the same. I noticed you already took care of hoisting  `ruby_engine_version`, `gem_spec`, and `native_platform`, but since the `payload` gets called for every request we might as well make sure to be as performant as possible 

The only values that might change are the ones in `state`

Reply:
This comment is not a blocker for this PR

Reply:
> We could probably hoist client_tracer altogether since between requests the values would be the same.

Some keys of `client_tracer` may change if `Datadog.configure` is called multiple times and we decide to start the component at another time than the first request.

Reply:
Not a blocker, but the only thing we are reading from the `configuration` is the service and the env. I'm not sure about those changes within a running app. I mean, technically they could, but I'm not they would.